src/Cache/RequestPolicy/DenyPost.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Drupal\graphql\Cache\RequestPolicy;
+
+use Drupal\Core\PageCache\RequestPolicyInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Contains a request policy that prevents caching of GraphQL POST requests.
+ */
+class DenyPost implements RequestPolicyInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function check(Request $request) {
+    if ($request->attributes->has('_graphql') && $request->getMethod() === Request::METHOD_POST) {
```

Comment:
should we change this to `$request->getMethod() !== Request::METHOD_GET`?

We don't want to cache PUT requests or similar.

This should really be the same as CommandLineOrUnsafeMethod?